### PR TITLE
Unify `O(sqrt(N))` `is_prime` functions under `project_euler`

### LINKED ISSUE #5434: Deduplicate repeated `is_prime` functions
We have a lot of `is_prime` (or similar) functions: https://github.com/TheAlgorithms/Python/search?p=4&q=is_prime, https://github.com/TheAlgorithms/Python/search?q=isPrime, [data_structures/hashing/number_theory/prime_numbers](https://github.com/TheAlgorithms/Python/blob/master/data_structures/hashing/number_theory/prime_numbers.py), etc. Shall we use one common function for that exactly identical `is_prime` calculation (which takes `O(sqrt(n))`)?

Candidates include:
* [maths/prime_check](https://github.com/TheAlgorithms/Python/blob/master/maths/prime_check.py)
* [maths/primelib](https://github.com/TheAlgorithms/Python/blob/master/maths/primelib.py)

### PR DESCRIPTION
**Pull request by elpaxoudis**
### Describe your change:

I changed the implementation of is_prime functions inside project_euler in order to have a unified implementation. There are some cases where the solution uses the Eratosthenes' sieve method. In some cases there is no specific gain from using that method so I changed it to the O(sqrt(n)) algorithm, but in other cases the sieve method is used in the core structure of the solution, hence I did not touch those.

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

fixes #5434

### CODE REVIEW

**Comment by elpaxoudis** [comment]
I think it is better to remove the assertion in all together since it doesn't have actual value for the implementation of the primality test algorithm and maybe it is a bit of confusing even. Should I do that? @poyea

**Comment by poyea** [comment]
@elpaxoudis Yeah, I think they can be removed, and the negative / type check can be performed when processing input (if any). If we want to handle the negative / type check in the function, I think it makes sense to raise an exception / an assertion failure, instead of return `False`.

File: project_euler/problem_003/sol1.py [hunk 1]
```diff
@@ -13,9 +13,11 @@
 import math
 
 
-def is_prime(num: int) -> bool:
-    """
-    Returns boolean representing primality of given number num.
+def is_prime(number: int) -> bool:
+    """Checks to see if a number is a prime in O(sqrt(n)).
+    A number is prime if it has exactly two factors: 1 and itself.
+    Returns boolean representing primality of given number num (i.e., if the
```
Comment (poyea) [inline project_euler/problem_003/sol1.py:19]:
```suggestion
    Returns boolean representing primality of given number (i.e., if the
```

File: project_euler/problem_007/sol1.py [hunk 2]
```diff
@@ -15,29 +15,37 @@
 from math import sqrt
 
 
-def is_prime(num: int) -> bool:
-    """
-    Determines whether the given number is prime or not
+def is_prime(number: int) -> bool:
+    """Checks to see if a number is a prime in O(sqrt(n)).
+    A number is prime if it has exactly two factors: 1 and itself.
+    Returns boolean representing primality of given number num (i.e., if the
```
Comment (poyea) [inline project_euler/problem_007/sol1.py:21]:
```suggestion
    Returns boolean representing primality of given number (i.e., if the
```

File: project_euler/problem_007/sol2.py [hunk 3]
```diff
@@ -11,22 +11,39 @@
...
 """
+import math
 
 
 def is_prime(number: int) -> bool:
-    """
-    Determines whether the given number is prime or not
+    """Checks to see if a number is a prime in O(sqrt(n)).
+    A number is prime if it has exactly two factors: 1 and itself.
+    Returns boolean representing primality of given number num (i.e., if the
```
Comment (poyea) [inline project_euler/problem_007/sol2.py:20]:
```suggestion
    Returns boolean representing primality of given number (i.e., if the
```

File: project_euler/problem_007/sol3.py [hunk 4]
```diff
@@ -16,20 +16,37 @@
 
 
 def is_prime(number: int) -> bool:
-    """
-    Determines whether a given number is prime or not
+    """Checks to see if a number is a prime in O(sqrt(n)).
+    A number is prime if it has exactly two factors: 1 and itself.
+    Returns boolean representing primality of given number num (i.e., if the
```
Comment (poyea) [inline project_euler/problem_007/sol3.py:21]:
```suggestion
    Returns boolean representing primality of given number (i.e., if the
```

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Comment by elpaxoudis** [comment]
@poyea Is there any other changes I should make?

**Review by poyea — APPROVED** [review]
It's been a while and this looks good to me. 👍

**Pull request merged**